docs/proposal-multi-user-approach.md

```diff
@@ -13,11 +12,276 @@ via a UI Server running as the suite owner for UI purposes meaning that, from
+    "<user2>": ["ADMIN", "!trigger", "!edit"],  # All Admin, not trigger, or edit
+    "<user3>": ["!ADMIN"]                       # No privileges (incl READ)
+}
+```
+
+### Example Site Configuration
+
+```python
+c.UIServer.authorisation = {
+    "<*>": {                    # For all ui-server owners,
+        "<*>": {                # Any authenticated user
+            "default": "READ",  # Will have default read-only access
+            "limit": "READ"     # All server owners default to being able to give away
+        },                      # a maximum of read privileges.
+        "<user1>": {            # user1
+            "default": "!ADMIN" # No privileges for all ui-server owners
+        },                      # No limit set, default level used as limit i.e. all
+    },                          # ui server owners default to be able to give away a
+                                # maximum of no privileges to user1
+
+    "<server_owner_1>": {       # For specific UI Server owner,
+        "<*>": {                # Any authenticated user
+            "default": "READ",  # Will have default read-only access
+            "limit": "ADMIN"    # server_owner_1 is able to give away
+        },                      # a maximum of admin privileges.
+    },
+    "<server_owner_2>": {       # Specific UI Owner
+        "<user2>": {            # Specific user2
+           "limit": "ADMIN"     # Can only be granted a maximum of Admin by
+        },                      # server_owner1, default access for user2 falls back to
+                                # standard READ only (if server_owner_2/user2 are
+                                # included in other auth config e.g. the top example),
+                                # or none if not in any other auth config sections.
+
+        "group:<groupA>": {     # group denoted with a `group:`
+            "default": "ADMIN"  # groupA has default ADMIN access to server_owner_2's
+        },                      # workflows
+    },
+    "group:<grp_of_svr_owners>":{         # Group of users who own UI Servers
+        "group: groupB": {
+            "default": "READ",            # can grant groupB users up to Admin
+            "limit": [                    # privileges, without stop and kill
+                "ADMIN", "!stop", "!kill" # operations
+            ]
+        },
+    }
+}
+```
+
+### Site vs. User Config Precedence, Defaults and Limits
+
+At CylcCon2020 is was agreed that a:
+
+> user can ramp up authorization levels as far as the site allows
+
+Site config `limit` will determine if a user has the privileges to give away access to their workflows. It will  it will also set a `default` access level. Users cannot raise access levels in their UI Server config for a given user or group, higher than those set in site config `limit`.
+
+If the site config does not set access for a given user or group then a UI Server will **not** be limited by site config.
```

Comment:
This seems at odds with the deny by default principle mentioned above. I think there should be no multi-user access possible unless permitted in the site config.

Reply:
Fair enough point, I will change to if a user is in the user config but not in the site config, we assume a user can not grant permission and deny?

Reply:
That's somewhat unfortunate, but I think we do need to do it.